File: web/src/api/miser/miser_loan_record.js
import service from '@/utils/request'

export const createMiserLoanRecord = (data) => {
  return service({
    url: '/miserLoanRecord/createMiserLoanRecord',
    method: 'post',
    data
  })
}

export const deleteMiserLoanRecord = (params) => {
  return service({
    url: '/miserLoanRecord/deleteMiserLoanRecord',
    method: 'delete',
    params
  })
}

export const deleteMiserLoanRecordByIds = (params) => {
  return service({
    url: '/miserLoanRecord/deleteMiserLoanRecordByIds',
    method: 'delete',
    params
  })
}

export const updateMiserLoanRecord = (data) => {
  return service({
    url: '/miserLoanRecord/updateMiserLoanRecord',
    method: 'put',
    data
  })
}

export const findMiserLoanRecord = (params) => {
  return service({
    url: '/miserLoanRecord/findMiserLoanRecord',
    method: 'get',
    params
  })
}

export const getMiserLoanRecordList = (params) => {
  return service({
    url: '/miserLoanRecord/getMiserLoanRecordList',
    method: 'get',
    params
  })
}

export const listMiserLoanNameList = () => {
  return service({
    url: '/miserLoanRecord/listMiserLoanNameList',
    method: 'get'
  })
}

export const getMiserLoanStatData = () => {
  return service({
    url: '/miserLoanRecord/getMiserLoanStatData',
    method: 'get'
  })
}

File: server/router/miser/miser_loan_record.go
package miser

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MiserLoanRecordRouter struct{}

func (s *MiserLoanRecordRouter) InitMiserLoanRecordRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	miserLoanRecordRouter := Router.Group("miserLoanRecord").Use(middleware.OperationRecord())
	{
		miserLoanRecordRouter.POST("createMiserLoanRecord", miserLoanRecordApi.CreateMiserLoanRecord)
		miserLoanRecordRouter.DELETE("deleteMiserLoanRecord", miserLoanRecordApi.DeleteMiserLoanRecord)
		miserLoanRecordRouter.DELETE("deleteMiserLoanRecordByIds", miserLoanRecordApi.DeleteMiserLoanRecordByIds)
		miserLoanRecordRouter.PUT("updateMiserLoanRecord", miserLoanRecordApi.UpdateMiserLoanRecord)
	}

	miserLoanRecordRouterWithoutRecord := Router.Group("miserLoanRecord")
	{
		miserLoanRecordRouterWithoutRecord.GET("findMiserLoanRecord", miserLoanRecordApi.FindMiserLoanRecord)
		miserLoanRecordRouterWithoutRecord.GET("getMiserLoanRecordList", miserLoanRecordApi.GetMiserLoanRecordList)
		miserLoanRecordRouterWithoutRecord.GET("listMiserLoanNameList", miserLoanRecordApi.ListMiserLoanNameList)
		miserLoanRecordRouterWithoutRecord.GET("getMiserLoanStatData", miserLoanRecordApi.GetMiserLoanStatData)
	}

	miserLoanRecordRouterWithoutAuth := PublicRouter.Group("miserLoanRecord")
	{
		_ = miserLoanRecordRouterWithoutAuth
	}
}
